src/main.rs

```diff
@@ -20,25 +20,36 @@ fn main() -> Result<(), Box<dyn Error>> {
         process::exit(0);
     }
 
-    let total = warnings.len();
+    let mut total = 0;
+    warnings.iter().for_each(|w| total += w.warnings.len());
```

Comment:
I think you can move the counting of totals to the `Output` struct 🤔

Reply:
This would be counting across `Output`s, since `run` gives a `Vec` of `Output`s.